pkg/dns/aws/dns.go

```diff
@@ -126,54 +135,57 @@ func NewProvider(config Config, operatorReleaseVersion string) (*Provider, error
 	tagConfig := aws.NewConfig()
 
 	// If the region is in aws china, cn-north-1 or cn-northwest-1, we should:
-	// 1. hard code route53 api endpoint to https://route53.amazonaws.com.cn and region to "cn-northwest-1" as route53 is not GA in AWS China,
-	//    and aws sdk didn't have the endpoint.
+	// 1. hard code route53 api endpoint to https://route53.amazonaws.com.cn and region to "cn-northwest-1"
+	//    as route53 is not GA in AWS China and aws sdk didn't have the endpoint.
 	// 2. use the aws china region cn-northwest-1 to setup tagging api correctly instead of "us-east-1"
 	switch region {
 	case endpoints.CnNorth1RegionID, endpoints.CnNorthwest1RegionID:
 		tagConfig = tagConfig.WithRegion(endpoints.CnNorthwest1RegionID)
-		r53Config = r53Config.WithRegion(endpoints.CnNorthwest1RegionID)
-		r53Config = r53Config.WithEndpoint("https://route53.amazonaws.com.cn")
+		r53Config = r53Config.WithRegion(endpoints.CnNorthwest1RegionID).WithEndpoint(chinaRoute53Endpoint)
+	case endpoints.UsGovEast1RegionID, endpoints.UsGovWest1RegionID:
+		// Route53 for GovCloud uses the "us-gov" region id:
+		// https://docs.aws.amazon.com/govcloud-us/latest/UserGuide/using-govcloud-endpoints.html
+		r53Config = r53Config.WithRegion(govCloudRoute53Region)
+		tagConfig = tagConfig.WithRegion(region)
```

Comment:
Note that I have not implemented the same tagging api region restrictions as standard aws. The PR needs to merge so @yunjiang29 can test against aws govcloud.